test/e2e/pivoting_test.go

```diff
@@ -281,3 +290,108 @@ func labelBMOCRDs(targetCluster framework.ClusterProxy) {
 		Expect(err).To(BeNil(), "Cannot label BMO CRDs")
 	}
 }
+
+func rePivoting() {
+
+	By("By removing Ironic deployment from target cluster")
+	removeIronicDeploymentOnTarget()
+
+	By("Reinstate Ironic containers and BMHD")
+	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
+	if ephemeralCluster == "kind" {
+		bmoPath := os.Getenv("BMOPATH")
+		ironicCommand := bmoPath + "/tools/run_local_ironic.sh"
+		cmd := exec.Command("sh", "-c", "export CONTAINER_RUNTIME=docker; "+ironicCommand)
+		stdoutStderr, err := cmd.CombinedOutput()
+		fmt.Printf("%s\n", stdoutStderr)
+		Expect(err).To(BeNil(), "Cannot run local ironic")
+	} else {
+
+		LogFromFile(filepath.Join(artifactFolder, "clusters", clusterName+"-pivoting", "clusterctl-init.log"))
+
+		By("Configure Ironic Configmap")
+		configureIronicConfigmap(true)
+
+		By("Install Ironic in the target cluster")
+		installIronicBMO(bootstrapClusterProxy, "true", "false")
+
+		By("Reinstate Ironic Configmap")
+		configureIronicConfigmap(false)
```

Comment:
what if we fail in any of these three phases in else block? Wouldn't if be good to have some phase specific error log here or the phases take care of this separately? By phase I mean, `configureIronicConfigmap, installIronicBMO and configureIronicConfigmap`

Reply:
Yes, the phases take care of error logs separately, the phases contain "Expect" function calls that are wrappers for the assertions and if any of them fails then we get a custom error message, the expected and the received value comparison and the full stack trace as output so we know exactly what failed.